src/components/utils/index.js

```diff
@@ -35,12 +35,20 @@ const enumerateDevicesInternal = (onSuccess, onError) => {
   }
 }
 
-export const checkIfHasWebcam = onResult => {
-  enumerateDevicesInternal(
-    devices => onResult( devices.some(device => device.kind === "videoinput") ),
-    () => onResult(false)
-  )
-}
+const checkDevicesInfo = checkFn =>
+  onResult =>
+    enumerateDevicesInternal(
+      devices => onResult(checkFn(devices)),
+      () => onResult(false)
+    )
+
+export const checkIfHasWebcam = checkDevicesInfo(
+  devices => devices.some(({ kind }) => kind === 'videoinput')
+)
+
+export const checkIfWebcamPermissionGranted = checkDevicesInfo(
+  devices => devices.some(({ kind, label }) => kind === 'videoinput' && !!label)
```

Comment:
`devices.some(({ kind, label }) => kind === 'videoinput'` seems to be repeated. maybe a function that returns the first webcam found can be created?

Reply:
Absolutely, much cleaner.